src/NumericalAlgorithms/LinearOperators/WeakDivergence.hpp

```diff
@@ -0,0 +1,191 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/Matrix.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Transpose.hpp"
+#include "DataStructures/Variables.hpp"
+#include "NumericalAlgorithms/LinearOperators/Divergence.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/Blas.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/Literals.hpp"
+#include "Utilities/StdArrayHelpers.hpp"
+
+/// \ingroup NumericalAlgorithmsGroup
+/// \brief Compute the weak form divergence of fluxes
+template <typename... FluxTags, size_t Dim>
+void weak_divergence(
+    const gsl::not_null<Variables<tmpl::list<Tags::div<FluxTags>...>>*>
+        divergence_of_flux,
+    const Variables<tmpl::list<FluxTags...>>& fluxes, const Mesh<Dim>& mesh,
+    const InverseJacobian<DataVector, Dim, Frame::Logical, Frame::Inertial>&
+        det_jac_times_inverse_jacobian) noexcept {
+  if (divergence_of_flux->number_of_grid_points() !=
+      fluxes.number_of_grid_points()) {
+    divergence_of_flux->initialize(fluxes.number_of_grid_points());
+  }
+
+  // Multiplies the flux be det_jac_time_inverse_jacobian.
+  const auto transform_to_logical_frame =
+      [&det_jac_times_inverse_jacobian, &fluxes](
+          auto tag_v,
+          const gsl::not_null<Variables<tmpl::list<Tags::div<FluxTags>...>>*>
+              result_buffer,
+          auto xi_or_eta_jacobian) noexcept {
+        using tag = tmpl::type_from<decltype(tag_v)>;
+        using div_tag = Tags::div<tag>;
+
+        auto& result = get<div_tag>(*result_buffer);
+        const auto& flux = get<tag>(fluxes);
+
+        for (size_t result_storage_index = 0;
+             result_storage_index < result.size(); ++result_storage_index) {
+          const auto result_tensor_index =
+              result.get_tensor_index(result_storage_index);
+          const auto x_tensor_index = prepend(result_tensor_index, 0_st);
```

Comment:
x_flux_tensor_index or flux_x_tensor_index ? (and similarly for the y and z components...)

Reply:
ping on this? you can do it when you squash; I'm happy with the other changes